### DISCUSSION TITLE
How to implement multiple themes for multiple pages with flowbite?

### DISCUSSION THREAD

**Discussion by designtology** [gatsbyjs/gatsby#38907] [Help]
I recently started with gatsby with a strapi cms. The theming was quite easy at first:
I created a theme object containing my buttons and other components and then created multiple themes out of it to implement a state that is responsible for switching between the themes via dropdown. So far so good.

Now I wanted to create a second page and I feel like my only option is to just copy and paste the exact same theme object (which is currently about 100 lines each theme) into the new page. So I thought: why not import from a file? Gatsby seems not to like that since I dont get a theme anymore. Even if I put the theme.js into the static files: No can do!

When logging the object into the console from the pages where I imported it I can see the content. But the <Flowbite> component seem to have an issue with imported files. What did I do wrong?

I know the theming in flowbite is highly experimental. But is Gatsby really not able to handle imported files? Or should I direct this issue to flowbite?

Thanks in advance.

**Comment by violy** [comment]
1. What do you meen by « _Themes_ » ? Multiples _layouts_ ? Styles variations … Please give more details on your meening of theme. 
2. Gatsby is using webpack to import modules. 
3. You should implement **Layout components** pattern to share same (or differents) layout(s), UIs, and styles on multiples pages → see [Gatsby Layout Components doc](https://www.gatsbyjs.com/docs/how-to/routing/layout-components/).
